### PR TITLE
docs: Add documentation for accessing the map object in <MapView>

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes
This PR some example documentation regarding accessing the [native maplibre-gl map object](https://maplibre.org/maplibre-gl-js-docs/api/map/) within the component that renders `<MapView>` by passing a ref, or from within a child component of `<MapView>` using the [useMap hook from react-map-gl](https://visgl.github.io/react-map-gl/docs/api-reference/use-map).

#### Issue #, if available
N/A

#### Description of how you validated changes
Visual validation

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated - N/A
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/geo/fragments/map-ref.react.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export default function MapWithRef() {
+  const mapRef = useRef();
+
+  const flyToMordor = useCallback(() => {
+    mapRef.current.flyTo({ center: [172.78, -42.28], zoom: 5 });
+  }, []);
+
+  return (
+    <>
+      <Button onClick={flyToMordor}>Fly, you fools!</Button>
```
Comment (thaddmt) [inline docs/src/pages/[platform]/components/geo/fragments/map-ref.react.mdx:21]: 🧙‍♂️🦅

**Review by thaddmt — APPROVED** [review]

File: docs/src/pages/[platform]/components/geo/react.mdx [hunk 2]
```diff
@@ -61,6 +61,20 @@ when the marker is clicked. The `offset` property can be used to prevent the pop
 _Note: Ensure `originalEvent.stopPropagation()` is called in the marker click handler. This allows the `showPopup`
 state to be handled by the component._
 
+### Animation and Native Map Methods
+
+You may want to access the [native maplibre-gl map object](https://maplibre.org/maplibre-gl-js-docs/api/map/) from within the component that renders `<MapView>` (to animate a viewport transition with `flyTo`, for example). To accomplish this, you can pass your own ref to `<MapView>` using React's `useRef` hook which will contain the `map` object:
```
Comment (reesscot) [inline docs/src/pages/[platform]/components/geo/react.mdx:66]: Add a link to React's useRef docs?

**Review by reesscot — APPROVED** [review]
LGTM

File: docs/src/pages/[platform]/components/geo/react.mdx (hunk 2 above)
Comment (slaymance) [reply docs/src/pages/[platform]/components/geo/react.mdx:66]: Yeah good call!

**Pull request merged**